### ISSUE TITLE
When using the sort plugin and moving a root node to another root spot, grid loses track of the moved node

### ISSUE THREAD

**Issue by ghost** [deitch/jstree-grid#84]
Grid was working and then when I added the sort plugin this bug pops up. It's only at root level. When you take a node and move it from a root level to another spot at root level it does not actually move because of the sort plugin. The problem is grid loses track of this and the grid below that item becomes offset and the item you attempted to move is then gone. If you do this kind of move on a child node, this isn't an issue and the grid is fine. If you move a node to a child or from a child to root it works like normal. It's only when moving from root to root.

**Comment by sfdctower** [comment]
@deitch was there any followup on this? I am having a similar issue.  

move_node.jstree works with jstree and jstreegrid but only when the "sort" plugin is omitted from jstree.

If I enable the "sort" plugin, jstreegrid columns do sort correctly, and seem to sort by first column ascending despite my server providing an ordering -_-.  The biggest challenge is that when I include the "sort" plugin with jstree+ jstreegrid then my root level nodes dont respond very well to move_node.jstree.  With "sort" enabled, move_node.jstree will move a node into other nodes but not between other nodes (not between nodes at the root, and not between nodes anywhere else).

It would be difficult for me to provide a fiddle, but would definitely troubleshoot this with someone if they wanted to reach out to me.

**Comment by sfdctower** [comment]
@deitch, although I'm not the original poster, I am 90% confident i have fixed this.  I will write back soon with my findings.

**Comment by deitch** [comment]
> was there any followup on this? I am having a similar issue

Not that I am aware of. There is no commercial side to jstree-grid - or company backing it - so my support rule of thumb is, if people create fiddles (or similar) so something is consistently recreatable, will address, else it isn't realistic.

> The challenge is that when I include the "sort" plugin with jstree+ jstreegrid then my root level nodes dont respond very well to move_node.jstree. With "sort" enabled, move_node.jstree will move a node into other nodes but not between other nodes (not between nodes at the root, and not between nodes anywhere else).

Still confused. It shouldn't be too hard to create a fiddle with a sample tree with mock data, and then instructions as to how to recreate?

**Comment by deitch** [comment]
>  I am 90% confident that I have fixed the problem I mentioned earlier here. I will write back with my findings ASAP

Thank you, that is great. Even better than recreating! :-)

**Comment by sfdctower** [comment]
@deitch, my problem is now resolved.

Findings:
Although I cannot speak to the original poster's issue,
my earlier claim here was not accurate, and certainly not a jstreegrid or jstree issue.  The issue I cited earlier in this conversation had indeed emerged whenever  "sort" plugin  was enabled, but turns out was not a problem with jstree or jstreegrid but was caused by poor coding on my part. That is to say after logging my variable assignments I realized that I was updating my server with "undefined" instead of the proper numerical data.position within move_node callback.
